### PR TITLE
devops: deploy to base mainnet

### PR DESCRIPTION
**Pull request by dcposch**
## Summary

- [x] Deploy and verify contracts
- [x] Pimlico USDC Base paymaster
- [x] Add second chain to chainConfig
- [x] Add `DAIMO_CHAIN` app build config to pick chain chainConfig

image

## Summary

- **old accounts pre-v8 dropped**
- **v8 includes homeChainId, homeCoinAddress.** this ensures it'll never be ambiguous whether an account is storage is mainnet or testnet.
- **added chain ID to API route.** this ensures that client and server are both testnet or both not

### Future improvement

in future, we can add side-by-side testnet support. this would mean something like:
- deep links include chain id (maybe optional, with a default, for shorter links)
- API supports multiple chains. where relevant, client passes chain ID in query
- (and potentially an in-app account switcher.)

### CODE REVIEW

File: packages/daimo-api/src/viemClient.ts [hunk 1]
```diff
@@ -27,12 +28,12 @@ import { jsonBigParse, jsonBigStringify } from "./jsonBig";
...
   // Connect to L1
   const l1Client = createPublicClient({
-    chain: mainnet,
+    chain: chainConfig.chainL1,
     transport: webSocket(process.env.DAIMO_API_L1_RPC_WS),
   });
 
   // Connect to L2
-  const chain = baseGoerli; // TODO: DAIMO_API_CHAIN once mainnet is supported
+  const chain = chainConfig.chainL2;
```
Comment (dcposch) [inline packages/daimo-api/src/viemClient.ts:36]: 🚀

File: packages/daimo-userop/src/nonce.ts [hunk 2]
```diff
@@ -77,3 +76,7 @@ export class DaimoNonceMetadata {
     return (hexNonceType + hexIdentifier) as Hex;
   }
 }
+
+function assert(condition: boolean) {
+  if (!condition) throw new Error("Assertion failed");
```
Comment (dcposch) [inline packages/daimo-userop/src/nonce.ts:81]: avoids a circular dependency.

`daimo-userop` only depends on `daimo-contracts` no, and both can be published to npm

**Comment by dcposch** [comment]
## Release testing checklist

### API and webapp

- [x] Push to prod
- [x] API deployed correctly
- [x] Webapp deployed correctly
- [x] Logs look clean

### iOS

Watch all of the following interactions closely for jank and UX regressions, not
just outright bugs.

- [x] Install from private TestFlight
- [x] ~Clear account (create one first, if necessary)~ skipped, mainnet
- [x] Create account
- [x] Send payment
- [ ] Create Payment Link, open in browser
- [ ] Cancel Payment Link
- [ ] Tap Payment Link, ensure it opens in-app
- [x] Create Request, open in browser
- [x] Fulfill request using Connect Wallet
- [ ] Add device + Use existing onboarding (test transactions work on both devices)
- [ ] Remove device
- [x] Open History
- [x] ~Tap transaction, view in block explorer~ broken due to incorrect paymaster special addr, fixed
- [x] Account screen: Send Debug Log
- [x] Debug log looks clean

### Android

- [ ] Install from Play Store closed track
- [ ] Clear account (create one first, if necessary)
- [ ] Create account
- [ ] Request faucet
- [ ] Notification appears on faucet send
- [ ] Send payment
- [ ] Create Payment Link, open in browser
- [ ] Cancel Payment Link
- [ ] Tap Payment Link, ensure it opens in-app
- [ ] Add device + Use existing onboarding (test transactions work on both devices)
- [ ] Remove device
- [ ] Create Request, open in browser
- [ ] Open History
- [ ] Tap transaction, view in block explorer
- [ ] Account screen: Send Debug Log
- [ ] Debug log looks clean

### ~Promote release~

This release is for internal testing only.

Production mainnet release coming shortly.

File: packages/contract/script/DeployEphemeralNotes.s.sol [hunk 3]
```diff
@@ -5,9 +5,10 @@ import "forge-std/Script.sol";
 import "../src/DaimoEphemeralNotes.sol";
 
 contract DeployEphemeralNotesScript is Script {
-    function run(IERC20 token) public {
+    function run() public {
         vm.startBroadcast();
-        new EphemeralNotes(token);
+        // DaimoEphemeralNotes defaults to USDC on Base.
+        new DaimoEphemeralNotes{salt: 0}(IERC20(address(0x0)));
```
Comment (nalinbhardwaj) [inline packages/contract/script/DeployEphemeralNotes.s.sol:11]: 🔥

File: packages/daimo-userop/src/daimoOpSender.ts [hunk 4]
```diff
@@ -12,56 +12,94 @@ import {
...
+    accountSigner: SigningCallback
+  ): Promise<DaimoOpSender> {
+    const { tokenAddress, tokenDecimals, paymasterAddress, chainL2 } =
+      Contracts.chainConfig;
 
-    this.notesAddress = _notesAddress;
+    let bundlerRpcUrl = process.env.DAIMO_BUNDLER_RPC || "";
+    if (bundlerRpcUrl === "" && chainL2.network === "base-goerli") {
+      bundlerRpcUrl = // Default to testnet
+        "https://api.pimlico.io/v1/base-goerli/rpc?apikey=70ecef54-a28e-4e96-b2d3-3ad67fbc1b07";
```
Comment (nalinbhardwaj) [inline packages/daimo-userop/src/daimoOpSender.ts:61]: looks like the API key Pimlico/Kristoff mentioned we should rotate? do you want to remove this before we go to mainnet?

**Review by nalinbhardwaj — APPROVED** [review]

File: packages/daimo-userop/src/daimoOpSender.ts (hunk 4 above)
Comment (dcposch) [reply packages/daimo-userop/src/daimoOpSender.ts:61]: It's been on GH for a few weeks

We have do a bit more than just rotate--need to proxy thru our API. Will do in a followup pr

**Pull request merged**